### PR TITLE
Adds video api endpoints

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

### CODE REVIEW

File: src/api/video/video.controller.js
Comment (spiray) [inline src/api/video/video.controller.js]: The convention is to use singular `Video` for the model name.

File: src/api/video/video.controller.js
Comment (spiray) [inline src/api/video/video.controller.js]: I don't like this formatting looks as is (not your fault). The pending PR #49 addresses this with the prettier implementation. We should wait to merge this PR until #49 is merged.

File: src/api/video/video.model.js
Comment (spiray) [inline src/api/video/video.model.js]: This should be `VideoSchema`.

File: src/helpers/youtubeFetcher.js
Comment (spiray) [inline src/helpers/youtubeFetcher.js]: This destructuring can be done in the argument in place of `item`.

File: src/helpers/youtubeFetcher.js
Comment (spiray) [inline src/helpers/youtubeFetcher.js:27]: We should handle this error a little differently.

**Review by spiray — CHANGES_REQUESTED** [review]
My initial thoughts. I will do a more thorough review tomorrow. Great work and thank you for working on this :)

File: src/helpers/youtubeFetcher.js
Comment (jbrundage) [reply src/helpers/youtubeFetcher.js:27]: That's out of the scope of this pull request

File: src/helpers/youtubeFetcher.js
Comment (ewenjo) [reply src/helpers/youtubeFetcher.js]: Don't forget to destruct `item.snippet.thumbnails` too

Comment (jbrundage) [reply src/helpers/youtubeFetcher.js]: I've already changed this line once before for destructuring's sake. Let's worry about bigger picture tasks and then we can do sweeps of ideal syntax changes later.

File: src/api/video/video.model.js
Comment (jbrundage) [reply src/api/video/video.model.js]: I agree

File: src/api/video/video.controller.js
Comment (jbrundage) [reply src/api/video/video.controller.js]: I agree that this format isn't ideal, but let's not let this hold up progression until we've delivered the MVP

Comment (spiray) [reply src/api/video/video.controller.js]: We aren't in a rush on these items as the frontend team is still working on their basic framework.

File: src/api/video/video.controller.js
Comment (jbrundage) [reply src/api/video/video.controller.js]: Totally agree... yeah I was kept 2nd guessing myself whether or not I should include the "s" in diff situations

File: src/helpers/youtubeFetcher.js
Comment (spiray) [reply src/helpers/youtubeFetcher.js]: See comment above. Lets get this right the first time so that we don't have to go back and fix things.

File: src/helpers/youtubeFetcher.js
Comment (spiray) [reply src/helpers/youtubeFetcher.js:27]: If you feel that it should be handled elsewhere you can create a Trello card for it. Thanks!

File: src/helpers/youtubeFetcher.js
Comment (jbrundage) [reply src/helpers/youtubeFetcher.js]: Then suggest a specific change and I'll make it

Comment (spiray) [reply src/helpers/youtubeFetcher.js]: This should do it -
```js
function transformYoutubeResponseData({ items }) {
  return items.map(({ snippet: { id: { videoID }, title, publishedAt, description, thumbnails: { high } } }) => {
    return {
      type: 'video',
      name: title,
      videoID: videoId,
      date: publishedAt,
      url: `https://www.youtube.com/watch?v=${videoId}`,
      thumbnail: high,
      title,
      description,
    };
  });
}
```

Comment (jbrundage) [reply src/helpers/youtubeFetcher.js]: I apologize, I misread your original comment. I see what you mean, but I think it's a bit harder to read when it's destructured in the argument. I think it's right as it is and because it follows a standard across its sibling files. If you think a change of those standards should be made (I could agree with that), it is out of scope of this pull request.

File: src/helpers/youtubeFetcher.js
Comment (ewenjo) [inline src/helpers/youtubeFetcher.js]: json() is a promise, and has to be "awaited"  `const data = await resp.json();`

Comment (jbrundage) [reply src/helpers/youtubeFetcher.js]: Good catch, thank you

File: src/helpers/youtubeFetcher.js
Comment (spiray) [reply src/helpers/youtubeFetcher.js]: You can alternatively do it like this but we should destructure it all the way regardless - 
```js
function transformYoutubeResponseData({ items }) {
  return items.map((item) => {
    const { snippet: { id: { videoID }, title, publishedAt, description, thumbnails: { high } } } = item;
    return {
      type: 'video',
      name: title,
      videoID: videoId,
      date: publishedAt,
      url: `https://www.youtube.com/watch?v=${videoId}`,
      thumbnail: high,
      title,
      description,
    };
  });
}
```

File: src/helpers/youtubeFetcher.js
Comment (jbrundage) [reply src/helpers/youtubeFetcher.js:27]: Is there a common way we should log JSON errors or something? `seed.js` and `localDevSeed.js` both use `catch(error) { console.log(error) }`. I thought about telling him he should change it to `console.error`, but I see `next(e)` all over the place... not sure what that is all about... but could you be more specific with change requests? Rather than leaving them up in the air?

Comment (spiray) [reply src/helpers/youtubeFetcher.js:27]: Yeah, this was an initial review. Ill explain our error handling structure in my review tomorrow. The main difference is that the error that we are logging are for local dev and the errors passed along to the error handling middleware are for sending as a response to the client.

**Comment by spiray** [comment]
@jbrundage The seedVideos function should be imported to the seed.js file and then awaited after the other 2 seed functions. Let me know if you need more information.

File: src/helpers/youtubeFetcher.js
Comment (spiray) [reply src/helpers/youtubeFetcher.js:27]: I am looking through this again. It seems that this is going to be a dev only function so we don't have to worry about the error handling. We are good with how it is now.

**Comment by spiray** [comment]
@jbrundage Can you also pull the updated develop branch, run the formatter and push up again?

**Comment by jbrundage** [comment]
@spiray @ewenjo I've amended this PR with the requested changes

File: src/api/video/video.controller.js
Comment (ewenjo) [inline src/api/video/video.controller.js:16]: The current prettier config allows this, perhaps we should set `prettierWidth` to something lower than 200. What do you think @spiray

**Review by ewenjo — APPROVED** [review]
Tests/linter passes, everything works. Requested changes have been made. Take a look at my comment, other than that, it looks good to me!

File: src/api/video/video.controller.js
Comment (spiray) [reply src/api/video/video.controller.js:16]: The default is 80 but I think that is too low. Should we do 160?

Comment (ewenjo) [reply src/api/video/video.controller.js:16]: It has to be at least 100/120 to affect this line. But setting it that low will also affect a lot of other files.

Comment (spiray) [reply src/api/video/video.controller.js:16]: I am personally fine with 200, I never had a problem with line lengths. If it becomes an issue for any contributors we can revisit this in the future.

Comment (ewenjo) [reply src/api/video/video.controller.js:16]: I agree, let's leave it for now.

**Review by spiray — APPROVED** [review]
Everything looks great!

**Pull request merged**
